src/bundle.rs

```diff
@@ -403,90 +404,131 @@ fn export_template(lems: &LemsFile, nml: &[String], bundle: &str) -> Result<()>
     }
 }
 
-#[derive(Debug)]
-struct Assign {
-    m: String,
-    g: Quantity,
-    e: Quantity,
+type Assign = (String, Quantity, Quantity);
+
+pub fn export_with_super_mechanisms(lems: &LemsFile, nml: &[String], bundle: &str) -> Result<()> {
+    let cells = read_cell_data(nml)?;
+    let chans = read_ion_channels(lems, nml)?;
+    let merge = build_super_mechanisms(&cells, &chans, lems)?;
+
+    for (id, cell) in merge {
+        for (reg, chan) in cell.channels {
+            let nmodl = nmodl::mk_nmodl(&chan)?;
+            let path = format!("{bundle}/cat/{id}_{reg}.mod");
+            info!("Writing Super-Mechanism NMODL for cell '{id}' region '{reg}' to {path:?}",);
+            write(&path, nmodl)?;
+        }
+        let path = format!("{bundle}/acc/{id}.acc");
+        info!("Writing Super Mechanism ACC to {path:?}");
+        write(&path, cell.decor.to_sexp())?;
+    }
+    Ok(())
+}
+
+pub struct Cell {
+    pub decor: Vec<Decor>,
+    pub channels: Vec<(String, Nmodl)>,
 }
 
-impl Assign {
-    fn new(m: &str, g: &str, e: &str) -> Result<Self> {
-        Ok(Self {
-            m: m.to_string(),
-            g: Quantity::parse(g)?,
-            e: Quantity::parse(e)?,
-        })
+pub fn build_super_mechanisms(
+    props: &Map<String, BiophysicalProperties>,
+    ins: &[Instance],
+    lems: &LemsFile,
+) -> Result<Map<String, Cell>> {
+    let sms = ion_channel_assignments(props, lems)?;
+    let dec = collect_decor(props, lems)?;
+    let mrg = merge_ion_channels(&sms, ins)?;
+    let mut result = Map::new();
+    for (cell, decor) in dec {
+        let channels = mrg[&cell].clone();
```

Comment:
This will panick!() if the cell has an empty membraneProperties (a case hit when comparing behaviours, where I remove all mechanisms and then add them one by one)

Reply:
True. The kind of damage C++ does to your brain, _there_ we'd just be getting the default.

Reply:
There, much nicer now.